src/go/k8s/pkg/resources/statefulset_update_test.go

```diff
@@ -71,7 +71,13 @@ func TestShouldUpdate_AnnotationChange(t *testing.T) {
 	}
 	stsWithAnnotation := sts.DeepCopy()
 	stsWithAnnotation.Spec.Template.Annotations = map[string]string{"test": "test2"}
-	ssres := StatefulSetResource{}
+	ssres := StatefulSetResource{
+		pandaCluster: &vectorizedv1alpha1.Cluster{
+			Status: vectorizedv1alpha1.ClusterStatus{
+				Restarting: false,
+			},
+		},
+	}
```

Comment:
But the unit test failed as there was missing panda cluster with the status. I believe it was panic due to nil dereference. It would be good to have checks for that.

Reply:
You cannot check for nil. It's not a type that can be nil.

```
pkg/resources/statefulset_update.go:578:30: invalid operation: r.pandaCluster.Status == nil (mismatched types "github.com/redpanda-data/redpanda/src/go/k8s/apis/vectorized/v1alpha1".ClusterStatus and untyped nil) (typecheck)
```

Reply:
Yes, but in the case of unit test pandaCluster was nil. 

In real life the pandaCluster should never be nil.